### PR TITLE
Fix backend to successfully create KIND_MODEL instances

### PR DESCRIPTION
**Pull request by tanmayv25**

### CODE REVIEW

File: src/backends/backend/triton_model_instance.cc [hunk 1]
```diff
@@ -79,10 +79,15 @@ TritonModelInstance::CreateInstances(
               model, group.name(), c, TRITONSERVER_INSTANCEGROUPKIND_GPU,
               device_id, instances));
         }
+      } else if (group.kind() == inference::ModelInstanceGroup::KIND_MODEL) {
+        RETURN_IF_ERROR(CreateInstance(
+            model, group.name(), c, TRITONSERVER_INSTANCEGROUPKIND_MODEL,
+            -2 /* device_id */, instances));
```
Comment (deadeyegoodwin) [inline src/backends/backend/triton_model_instance.cc:85]: Why do you use -2 for device ID? It should be ignored for KIND_MODEL.

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: src/backends/backend/triton_model_instance.cc (hunk 1 above)
Comment (tanmayv25) [reply src/backends/backend/triton_model_instance.cc:85]: Just following the convention here: https://github.com/triton-inference-server/tensorflow_backend/blob/main/src/tensorflow.cc#L1083
This indeed will be ignored for kind_model here: https://github.com/triton-inference-server/server/blob/master/src/backends/backend/triton_model_instance.cc#L43-L50

File: qa/L0_model_config/autofill_noplatform_success/tensorflow_savedmodel/kind_model_config/expected.3 [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: "kind_model_config"
```
Comment (Tabrizian) [inline qa/L0_model_config/autofill_noplatform_success/tensorflow_savedmodel/kind_model_config/expected.3:1]: License

File: src/backends/backend/triton_model_instance.cc (hunk 1 above)
Comment (deadeyegoodwin) [reply src/backends/backend/triton_model_instance.cc:85]: The -2 is tensorflow specific. Here in this common code you should just use 0 (zero).

Comment (tanmayv25) [reply src/backends/backend/triton_model_instance.cc]: Done

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**